tests/TestCase/ORM/TableTest.php

```diff
@@ -1181,6 +1181,15 @@ public function testFindApplyOptions(): void
         $table->find('all', $options);
     }
 
+    /**
+     * Tests that extra arguments are passed to finders.
+     */
+    public function testFindTypedParameters(): void
+    {
+        $author = $this->getTableLocator()->get('Authors')->find('WithIdArgument', [], 2)->first();
```

Comment:
Having to pass an empty array always when I don't want to use options but only custom args feels clunky. We could drop the `$options` arg and if the 1st variadic arg is an array treat it as options array but that it becomes a "magic" argument which isn't nice either.

Reply:
Though I can avoid the empty array using named params, `find(type: 'withIdArgument', id: 2)`.

Reply:
I tried this but this is where `getOptions()` comes in. We currently filter out query-specific options before passing the $options array to the finder.

This isn't straight forward when supporting named and non-named arguments together.

Reply:
I think I can accomplish this using references.

@markstory are you ok with dropping the `array $options` parameter and making it optional for finders?

Reply:
> are you ok with dropping the array $options parameter and making it optional for finders?

That could work. How would users that rely on $options update their code?

Reply:
If they pass an array as positional argument #2 or a named `$options` argument, it would work normally.

Reply:
That's more or less what I was trying to do in #12787. I'm OK with dropping the requirement of passing an empty array, but not sure about special treatment of array arguments. It may not be clear that if a first custom finder argument is expected to be an array then it should be passed second after an empty array.

Reply:
Don't really have to say much about this other than LGTM and I am fine with this change in 5.x 👍🏻